plugins/multifundchannel.c

```diff
@@ -0,0 +1,1939 @@
+	}
+
+	plugin_log(mfc->cmd->plugin, LOG_DBG,
+		   "mfc %"PRIu64": sendpsbt.", mfc->id);
+
+	req = jsonrpc_request_start(mfc->cmd->plugin, mfc->cmd,
+				    "sendpsbt",
+				    &after_sendpsbt,
+				    &mfc_forward_error,
+				    mfc);
+	json_add_psbt(req->js, "psbt", mfc->psbt);
+	return send_outreq(mfc->cmd->plugin, req);
+}
+
+static struct command_result *
+multifundchannel_finished(struct multifundchannel_command *mfc);
+
+static struct command_result *
+after_sendpsbt(struct command *cmd,
+	       const char *buf,
+	       const jsmntok_t *result,
+	       struct multifundchannel_command *mfc)
+{
+	const jsmntok_t *tx_tok;
+	const jsmntok_t *txid_tok;
+
+	plugin_log(mfc->cmd->plugin, LOG_DBG,
+		   "mfc %"PRIu64": sendpsbt done.", mfc->id);
+
+	tx_tok = json_get_member(buf, result, "tx");
+	if (!tx_tok)
+		plugin_err(cmd->plugin,
+			   "sendpsbt response has no 'tx': %.*s",
+			   json_tok_full_len(result),
+			   json_tok_full(buf, result));
+	mfc->final_tx = json_strdup(mfc, buf, tx_tok);
+
+	txid_tok = json_get_member(buf, result, "txid");
+	if (!txid_tok)
+		plugin_err(cmd->plugin,
+			   "sendpsbt response has no 'txid': %.*s",
+			   json_tok_full_len(result),
+			   json_tok_full(buf, result));
+	mfc->final_txid = json_strdup(mfc, buf, txid_tok);
+
+	/* PSBT is no longer something we need to clean up.  */
+	mfc->psbt = tal_free(mfc->psbt);
+
+	return multifundchannel_finished(mfc);
+}
+
+/*---------------------------------------------------------------------------*/
+/*~
+And finally we are done, after a thousand lines or so of code.
+*/
+
+static struct command_result *
+multifundchannel_finished(struct multifundchannel_command *mfc)
+{
+	unsigned int i;
+	struct json_stream *out;
+
+	plugin_log(mfc->cmd->plugin, LOG_DBG,
+		   "mfc %"PRIu64": done.", mfc->id);
+
+	out = jsonrpc_stream_success(mfc->cmd);
+	json_add_string(out, "tx", mfc->final_tx);
+	json_add_string(out, "txid", mfc->final_txid);
+	json_array_start(out, "channel_ids");
+	for (i = 0; i < tal_count(mfc->destinations); ++i) {
+		json_object_start(out, NULL);
+		json_add_node_id(out, "id", &mfc->destinations[i].id);
+		json_add_string(out, "channel_id", mfc->destinations[i].channel_id);
+		json_add_num(out, "outnum", mfc->destinations[i].outnum);
+		json_object_end(out);
+	}
+	json_array_end(out);
+
+	return mfc_finished(mfc, out);
+}
+
+/*-----------------------------------------------------------------------------
+Command Entry Point
+-----------------------------------------------------------------------------*/
+
+/* Entry function.  */
+static struct command_result *
+json_multifundchannel(struct command *cmd,
+		      const char *buf,
+		      const jsmntok_t *params)
+{
+	const jsmntok_t *json_destinations;
+	const char *feerate_str;
+	u32 *minconf;
+	const char *utxos_str;
+
+	struct multifundchannel_command *mfc;
+
+	struct command_result *result;
+
+	if (!param(cmd, buf, params,
+		   p_req("destinations", param_tok, &json_destinations),
```

Comment:
Always prefer to do inline parsing where possible (rahter than using param_tok and parsing later).  That way the "check" command is far more effective (which always exits this function after param).

Of course, IIRC we haven't implemented check for plugins, but the theory applies!